soc/arc/snps_nsim/CMakeLists.txt

```diff
@@ -23,6 +23,17 @@ else()
 			       -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
 			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
 
+  zephyr_compile_options_ifdef(CONFIG_SOC_NSIM_EM11D -arcv2em -core3 -Xdiv_rem=radix2
+			       -Xmpy_option=mpyd -Xbitscan -Xswap -Xbarrel_shifter
+			       -Xshift_assist -Xdsp2 -Xdsp_complex
+			       -Xdsp_divsqrt=radix2 -Xdsp_itu -Xdsp_accshift=full
+			       -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
+			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
+
+  if(CONFIG_SOC_NSIM_EM11D)
+    set(ENV{EM11D_DSP_OPTIONS} "-Xxy;-Xagu_large;-Xdsp_ctrl=postshift,noguard,convergent;-Hfxapi")
```

Comment:
Let's use more generic naming so we can avoid extra `if`'s in test's cmake when we add more ARC DSP-enabled configurations. Like `EM11D_DSP_OPTIONS` -> `Z_ARC_DSP_OPTIONS`.